File: algorithms/fibbonaci/fibbonaci.cpp
int fibb_iterate(int n) {
    if (n <= 1) {
        return n;
    } else {
        int num1 = 0, num2 = 1;
        for (int i = 2; i <= n; ++i) {
            int result = num1 + num2;
            num1 = num2;
            num2 = result;
            std::cout << num2 << std::endl;
        }
        return num2;
    }
}

int fibb_recurrence(int n) {
    if (n <= 1) {
        return n;
    } else {
        return fibb_recurrence(n - 1) + fibb_recurrence(n - 2);
    }
}

File: algorithms/fibbonaci/fibbonaci.py
# Fibbonaci sequence using iteration
def fibb_iterate(n) -> int:
    if n <= 1:
        return n
    else:
        num1, num2 = 0, 1
        for i in range(0, n):
            result = num1 + num2
            num1 = num2
            num2 = result
            return num2


# Fibbonaci sequence using recursion
def fibb_rec(n) -> int:
    if n <= 1:
        return n
    else:
        return fibb_rec(n-1) + fibb_rec(n-2)